gooey/gui/components/menubar.py

```diff
@@ -78,4 +80,23 @@ def spawnAboutDialog(self, item, *args, **kwargs):
             if field in item:
                 getattr(about, method)(item[field])
 
-        three_to_four.AboutBox(about)
\ No newline at end of file
+        three_to_four.AboutBox(about)
+
+    def spawnPrintDialog(self, item, *args, **kwargs):
+        fonts = {
+            "DEFAULT" : wx.FONTFAMILY_DEFAULT,
+            "DECORATIVE" : wx.FONTFAMILY_DECORATIVE,
+            "ROMAN" : wx.FONTFAMILY_ROMAN,
+            "SCRIPT" : wx.FONTFAMILY_SCRIPT,
+            "SWISS" : wx.FONTFAMILY_SWISS,
+            "MODERN" : wx.FONTFAMILY_MODERN,
+            "TELETYPE" : wx.FONTFAMILY_TELETYPE,
+            "MAX" : wx.FONTFAMILY_MAX
+        }
+        printer = wx.richtext.RichTextPrinting()
+        buf = wx.richtext.RichTextBuffer()
+        buf.BeginFont(wx.Font(item.get("size",10),
+            fonts[item.get("font","TELETYPE")],
+            wx.FONTSTYLE_NORMAL,wx.FONTWEIGHT_NORMAL))
+        buf.AddParagraph(self.application.console.getText())

```

Comment:
(thinking out loud here) I wonder if there's a better approach to this. As is, it feels weird that a Menu Bar would know, and have concrete reference to, the main application (_and_ drill down into the _application's_ dependency to get the console text). It's a bit Demeter-y smell wise. 

Maybe the dependencies should go the other way, so the `application` holds a dependency on a `PrintManager` kind of class that houses all the "how to" related to Printing, and then all this Menu fella would be responsible for was dispatching an event via the PubSub mechanisms?